components/board/pi/impl/board.go

```diff
@@ -112,20 +112,24 @@ func NewPigpio(ctx context.Context, cfg *genericlinux.Config, logger golog.Logge
 
 	instanceMu.Lock()
 	logger.Info("initializing pigpio C library")
-	resCode = C.gpioInitialise()
-	if resCode < 0 {
-		pigpioInitialized = false
-		instanceMu.Unlock()
-		// failed to init, check for common causes
-		_, err := os.Stat("/sys/bus/platform/drivers/raspberrypi-firmware")
-		if err != nil {
-			return nil, errors.New("not running on a pi")
-		}
-		if os.Getuid() != 0 {
-			return nil, errors.New("not running as root, try sudo")
+
+	// if pigpio is not initialized, only then we initialize it.
+	if !pigpioInitialized {
+		resCode = C.gpioInitialise()
+		if resCode < 0 {
+			instanceMu.Unlock()
+			// failed to init, check for common causes
+			_, err := os.Stat("/sys/bus/platform/drivers/raspberrypi-firmware")
+			if err != nil {
+				return nil, errors.New("not running on a pi")
+			}
+			if os.Getuid() != 0 {
+				return nil, errors.New("not running as root, try sudo")
+			}
+			return nil, errors.Errorf("gpioInitialise failed with code: %d", resCode)
 		}
-		return nil, errors.Errorf("gpioInitialise failed with code: %d", resCode)
 	}
+
```

Comment:
[major] any way you could test this out in a software test? You'd have to run it on a pi (`make test-pi` on the pi you're developing on from within an rdk folder).

Reply:
Just finished running the pi tests. If server is already running and we try to initialize the gpio, we would get -1 error "gpioInitialise failed" and proceed to the next thing. The only way to get around this behavior is to stop the server and start it again. 
Are we ok with this behavior?

Reply:
This is okay and expected. The pigpio library only allows one process to access the underlying hardware, so two processes trying to access it simultaneously is not something we can support.